mlflow/server/js/src/shared/web-shared/model-trace-explorer/ModelTraceHeaderDetails.tsx

```diff
@@ -0,0 +1,217 @@
+
+    if (rootSpan) {
+      const startTime = getModelTraceSpanStartTime(rootSpan);
+      const endTime = getModelTraceSpanEndTime(rootSpan);
+
+      if (startTime && endTime && endTime > startTime) {
+        return `${((endTime - startTime) / 1000000).toFixed(1)}s`;
+      }
+    }
+
+    return undefined;
+  }, [spans]);
+
+  const latency = calculateLatency();
+
+  const idLabel = intl.formatMessage({
+    defaultMessage: 'ID',
+    description: 'Label for the ID section',
+  });
+
+  const tokenCountLabel = intl.formatMessage({
+    defaultMessage: 'Token count',
+    description: 'Label for the token count section',
+  });
+
+  const latencyLabel = intl.formatMessage({
+    defaultMessage: 'Latency',
+    description: 'Label for the latency section',
+  });
+
+  const tagsLabel = intl.formatMessage({ defaultMessage: 'Tags', description: 'Label for the tags section' });
+
+  const notificationSuccessMessage = intl.formatMessage({
+    defaultMessage: 'Copied to clipboard',
+    description: 'Success message for the notification',
+  });
+
+  const getComponentId = useCallback((key: string) => `${BASE_TAG_COMPONENT_ID}.tag-${key}`, []);
+
+  const handleTagClick = (text: string) => {
+    navigator.clipboard.writeText(text);
+  };
+
+  const getTruncatedLabel = (label: string) => truncateToFirstLineWithMaxLength(label, 40);
+
+  const handleCopy = useCallback(() => {
+    setShowNotification(true);
+    setTimeout(() => setShowNotification(false), 2000);
+  }, []);
+
+  return (
+    <>
+      <div css={{ display: 'flex', flexDirection: 'row', gap: theme.spacing.md, flexWrap: 'wrap' }}>
+        {modelTraceId && (
+          <ModelTraceHeaderMetricSection
+            label={idLabel}
+            value={modelTraceId}
+            tagKey={modelTraceId}
+            color="pink"
+            getTruncatedLabel={getTruncatedLabel}
+            getComponentId={getComponentId}
+            onCopy={handleCopy}
+          />
+        )}
+        {totalTokens && (
+          <ModelTraceHeaderMetricSection
+            label={tokenCountLabel}
+            value={totalTokens.toString()}
+            tagKey="token-count"
+            color="default"
+            getTruncatedLabel={getTruncatedLabel}
+            getComponentId={getComponentId}
+            onCopy={handleCopy}
+          />
+        )}
+        {latency && (
+          <ModelTraceHeaderMetricSection
+            label={latencyLabel}
+            value={latency}
+            tagKey="latency"
+            color="default"
+            getTruncatedLabel={getTruncatedLabel}
+            getComponentId={getComponentId}
+            onCopy={handleCopy}
+          />
+        )}
+        <div
+          css={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+            flexDirection: 'row',
+            gap: theme.spacing.sm,
+          }}
+        >
+          <Typography.Text size="md" color="secondary">
+            {tagsLabel}
```

Comment:
nit: if it's just a single label, generally we prefer to inline it
```suggestion
            <FormattedMessage
              defaultMessage="Tags"
              description="Label for the tags section"
            />
```

Reply:
OK, sure.